docs/sequencer/sequencer.md

```diff
@@ -134,49 +137,106 @@ def g1_powers_check(contribution: BatchContribution) -> bool:
             if bls.pairing(bls.G1.g1, g2_power) != bls.pairing(g1_power, bls.G2.g2):
                 return False
     return True
 ```
 
-## Updating the transcript
+### Verifying a `Contribution`
+
+The sequencer MUST verify all above checks:
+
+```python
+def verify_contribution(contribution: contribution, identity: str) -> bool:
+    for check in [schema_check, parameter_check, subgroup_checks, tau_update_check, g1_powers_check, g2_powers_check]:
+        if not check(contribution):
+            return False
+    return True
+```
 
-Once the sequencer has performed the above Verification checks, they MUST update the transcript.
+## Signature Verification & Pruning
+
+The sequencer MUST verify and prune the signatures in the contributions.
+
+- __BLS Signature Verification__ - Verify that the participant correctly signed their contributor identity. It is assumed that signature encoding errors return `False`.
+
+```python
+def bls_signature_check(contribution: Contribution, identity: str) -> bool:
+    for contribution in transcript.contributions:
+        pubkey = contribution.pot_pubkey
+        message = identity
+        signature = contribution.bls_signature
+        if not bls.Verify(pubkey, message, signature):
+            return False
+    return True
+```
+
+- __ECDSA Signature Verification__ - Verify that the participant correctly signed their PoT Pubkeys. It is assumed that signature encoding errors return `False`.
+
+```python
+def ecdsa_signature_check(contribution: Contribution, identity: str) -> bool:
+    ecdsa_signature = contribution.ecdsa_signature
+    typed_data = contribution_to_typed_data_str(contribution)
+    message = encode_structured_data(typed_data)  # Encode Data as per EIP 712
+    recovered_address = recover_message(message, contribution.ecdsa_signature)  # Address recovery
+    return recovered_address == identity[4:]  # Strip off leading 'eth|' padding from id & verify
+```
+
+### Signature Pruning
+
+The sequencer MUST prune invalid signatures from the contribution.
+
+```python
+def prune_invalid_signatures(contribution: Contribution, identity: str) -> Contribution:
+    if not bls_signature_check(contribution, identity):
+        for contribution in contribution:
+            contribution.pot_pubkey = ''
+    if identity[:4] != 'eth|' or not ecdsa_signature_check(contribution, identity):
+        contribution.bls_signature = ''
+    return contribution
+```
+
+## Updating the transcript
 
+Once the sequencer has performed the above Verification checks and assuming they all passed, and the signatures have been pruned as above, they MUST update the transcript. Thy do so by adding the `contribution` to the `transcript`. The `identity` is a string that represents the user. It is generated by talking the Ethereum address or GitHub account from the users' login session and feeding them into `eth_address_to_identity` or `github_handle_to_identity` respectively.
```

Comment:
Verification -> verification
talking -> taking

Reply:
ACK ACK, fixed in https://github.com/ethereum/kzg-ceremony-specs/pull/33/commits/d9c241c7b0dad8f5af0de2ec03061b45a7d3a84a